src/number/intFromString.ts

```diff
@@ -17,48 +20,64 @@ import * as G from 'io-ts/Guard'
 import * as TD from 'io-ts/TaskDecoder'
 import * as t from 'io-ts/Type'
 import * as N from 'fp-ts/number'
-import { flow } from 'fp-ts/function'
-import { Type as Type_ } from 'io-ts'
+import * as Str from 'fp-ts/string'
+import { pipe } from 'fp-ts/function'
+import { failure, Type as Type_ } from 'io-ts'
 
 import * as Arb from '../internal/ArbitraryBase'
 import * as int from './int'
 
+/**
+ * Represents all bases numbers can encode to using `Number.prototype.toString(base)`
+ * which have a specified prefix. This is necessary for decoder / encoder laws to hold.
+ *
+ * @since 1.0.0
+ */
+export type IntFromStringParams = {
+  readonly encodeToBase?: 2 | 8 | 10 | 16
```

Comment:
:thinking:  why limit to these? `parseInt` and `toString` support bases 2-36

Reply:
I had it as that union originally, but realized without being able to decode from a specific radix it was unlawful to round trip.

That said, it's technically still unlawful for exponentials and big ints, but I'm not sure how to handle those

Reply:
The reason I'm using `Number > isInt` instead of `parseInt(radix) > isInt`, is that parseInt would convert `1.1` to `1`, and I don't think that's desired.  Though I'm open to being persuaded on that front

Reply:
it's not the introduction of the base that makes it unlawful though is it?

Might want to just leave a note in the doc comment and if anyone wants to do that they can open an issue to request the feature

Reply:
I think it's because you can't decode anything other than binary, octal, decimal, or hexadecimal when using the `Number` constructor.  Though it would work with parseInt... Actually, I think it'd just be better to use parseInt.  Let me try that real quick.  Because `parseInt("1gm9", 30)` works, but `Number("1gm9")` doesn't.  So I actually think it's an improvement

Reply:
It's because `Number` can only parse something that has a prefix.  And `parseInt` has problems, like consider the following:

```ts
parseInt("1gm9", 16) === 1 
parseInt("1gm9", 30) // works as expected
```